File: src/Domain/Entities/Course.cs
namespace SkillSphere.Domain.Entities;

public class Course : BaseAuditableEntity
{
    public string? Title { get; set; }
    public string? Description { get; set; }
    public string? CoverImageRelativePath { get; set; }

    public bool? IsPublished { get; set; } = false;
    public float Price { get; set; }

    public int Likes { get; set; } = 0;
    
    public IList<Chapter> Chapters { get; init; } = new List<Chapter>();

    public IList<CourseCategory> Categories { get; init; } = new List<CourseCategory>();
    
    public IList<UserCourse> UserCourses { get; init; } = new List<UserCourse>();
    
    public IList<WishListItem> WishList { get; init; } = new List<WishListItem>();
}

File: src/UI/src/app/modules/management/models/course.ts
export interface Course {
    id: string; 
    title: string; 
    description: string; 
    coverImageRelativePath: string; 
    isPublished:boolean; 
    price: number; 
    likes: number;
    isLiked?: boolean;
    isPurchased?: boolean;
    isOwned?: boolean;
    categories: Categories[]; 
    chapters: Chapter[]; 
    authorName: string;
  }

export interface Categories {
    categoryId: string; 
    name: string; 
  }

  export interface Chapter {
    id: string;
    title: string; 
    description: string;
    videoURL: string; 
    position:number;
    isFree:boolean;
  }
  
  // price tier list 
  // Free
  // 19.99 (tier 1)
  // 29.99 (tier 2)
  // 30.99 (tier 3)